src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -83,6 +98,10 @@ public CSharpType(TypeProvider implementation, string ns, string name, bool isVa
         public string Name { get; }
         public bool IsValueType { get; }
         public bool IsEnum { get; }
+        public bool IsLiteral { get; set; }
```

Comment:
This type should be immutable.  It would be chaotic if each time you read these values they could change.